### PR TITLE
Enhance map rename key

### PR DESCRIPTION
**Pull request by boxxxie**

### CODE REVIEW

**Review by olafura — CHANGES_REQUESTED** [review]
Maybe do this with a option?

**Review by olafura — APPROVED** [review]

File: lib/kitchen_sink/map.ex [hunk 1]
```diff
@@ -136,41 +136,41 @@ defmodule KitchenSink.Map do
...
-  def rename_key(%{} = map, current_key, new_key) do
+  def rename_key(%{} = map, current_key, new_key, options) do
     {popped, popped_map} = Map.pop(map, current_key)
 
-    case popped do
-      nil -> map
-      _ -> Map.put_new(popped_map, new_key, popped)
+    case {popped, Keyword.get(options, :overwrite)} do
+      {nil, true} -> map
+      {nil, false} -> map
```
Comment (olafura) [inline lib/kitchen_sink/map.ex]: Maybe use do `{nil, check} is_boolean(check) -> map`

**Review by olafura — CHANGES_REQUESTED** [review]

File: lib/kitchen_sink/map.ex (hunk 1 above)
Comment (boxxxie) [reply lib/kitchen_sink/map.ex]: probably faster not to do that

Comment (boxxxie) [reply lib/kitchen_sink/map.ex]: fixed

**Review by olafura — APPROVED** [review]

**Review by robertpeacock22 — CHANGES_REQUESTED** [review]
I would love to see a test for the case where `overwrite: false`

**Review by robertpeacock22 — APPROVED** [review]

**Pull request merged**